File: sdk/node/src/api/learning.path.courses/learning.path.courses.model.ts

import { IsString, IsNumber, IsOptional, IsBoolean, IsUUID } from 'class-validator';

export class LearningPathCoursesCreateModel {
    @IsOptional()
    @IsUUID()
    id?: string;  

}

export class LearningPathCoursesUpdateModel {

}

export class LearningPathCoursesSearchParams {
    @IsString()
    name: string;
}

export class ApiResponse<T> {
    Data: T;

    @IsOptional()
    @IsString()
    Message?: string;

    @IsOptional()
    @IsBoolean()
    Success?: boolean;
}

File: sdk/python/src/lmsservice_db/api/learning_path_courses/learning_path_courses_model.py

from dataclasses import dataclass, field
from typing import Optional, Dict

@dataclass
class LearningPathCoursesCreateModel:


@dataclass
class LearningPathCoursesUpdateModel:


@dataclass
class LearningPathCoursesSearchParams:


@dataclass
class ApiResponse:
    Data: Dict
    Message: Optional[str] = field(default=None, metadata={"min_length": 2, "max_length": 500})
    Success: Optional[bool] = None
